_posts/2023-08-29-cartsguru.md

```diff
@@ -0,0 +1,122 @@
+
+
+## Summary
+
+* **CVE ID**: [CVE-2023-39642](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-39642)
+* **Published at**: 2023-08-29
+* **Platform**: PrestaShop
+* **Product**: cartsguru
+* **Impacted release**: <= 2.4.2 [considered to be "truly" fixed on 2.4.3 - see note below]
+* **Product author**: Carts Guru
+* **Weakness**: [CWE-89](https://cwe.mitre.org/data/definitions/89.html)
+* **Severity**: critical (9.8)
+
+## Description
+
+The method `CartsGuruCatalogModuleFrontController::display()` and the ajax script controllers14/catalog.php has sensitive SQL calls that can be executed with a trivial http call and exploited to forge a SQL injection.
+
+Note : The author has deleted from its module the files that have been suffering from critical vulnerabilities for months, BUT did not set them to be "auto-deleted" during upgrades. Therefore, there are likely merchants out there with older versions who have updated their modules thinking they are safe. However, there is nothing safe about this since past upgrades do not auto-delete the implicated files. To ensure everyone has a "safe version", we decided to mark all versions up to 2.4.3 as impacted by this issue.
+
+**WARNING** : This exploit is actively used to deploy webskimmer to massively steal credit cards. 
+
+One of these exploits uses a PrestaShop front controller and most attackers can conceal the module controller's path during the exploit, so you will never know within your conventional frontend logs that it exploits this vulnerability. **You will only see "POST /" inside your conventional frontend logs.** Activating the AuditEngine of mod_security (or similar) is the only way to get data to confirm this exploit.
+
+## CVSS base metrics
+
+* **Attack vector**: network
+* **Attack complexity**: low
+* **Privilege required**: none
+* **User interaction**: none
+* **Scope**: unchanged
+* **Confidentiality**: high
+* **Integrity**: high
+* **Availability**: high
+
+**Vector string**: [CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H](https://nvd.nist.gov/vuln-metrics/cvss/v3-calculator?vector=AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H)
+
+## Possible malicious usage
+
+* Obtain admin access
+* Remove data from the associated PrestaShop
+* Copy/paste data from sensitive tables to FRONT to expose tokens and unlock admins's ajax scripts
+* Rewrite SMTP settings to hijack emails
+
+
+## Proof of concept
+
+
+```bash
+curl -v 'https://preprod.XX/modules/cartsguru/controllers14/catalog.php?cartsguru_catalog_limit=1;select(0x73656C65637420736C656570283432293B)INTO@a;prepare`b`from@a;execute`b`;--'
+curl -v 'https://preprod.XX/?fc=module&module=cartsguru&controller=catalog&cartsguru_catalog_limit=1;select(0x73656C65637420736C656570283432293B)INTO@a;prepare`b`from@a;execute`b`;--'
+```
+
+## Patch from 1.4.19
+
+These files does not exist if you started working with Cartsguru beyond 2.X versions BUT if you worked with them BEFORE 2.X versions - you should own it.
```

Comment:
These files do not exist if you started working with Cartsguru beyond version 2.X BUT if you worked with them BEFORE 2.X versions - you should own it.